### PR TITLE
MB-13693 - bug: SIT module not showing

### PR DESCRIPTION
**Pull request by scottmries**
## Summary

This change primarily updates the `ShipmentSITDisplay` component to appear whenever there is SIT, not just when its dates are passed. A number of console errors had been present, so I also added falsy checks where they had been occurring.

Please pay particular attention to how the start and end dates are derived from `MTOServiceItems` when `sitStatus` is falsy; this seemed to be the best way to derive that data, but I want to ensure that it's accurate.

I used [this Figma](https://www.figma.com/file/5AhuEkaH3bJVrtffU3wp22/Milmove-Office-App?node-id=10285-112280&t=uGp8B0Jevqof1Q82-0) as a point of reference.

I also created a move with the code `SITFUT` ("future SIT") to test this behavior.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Access the office app
2. Login as a TOO
3. Select each of the following moves, then click "Move task order," and ensure that the panel displays for each:
- `PENDNG`
- `SITEXT`
- `SITFUT`

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/main/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

## Screenshots

`SITEXT`
SITEXT

`PENDNG`
PENDNG

`SITFUT`
SITFUT

### CODE REVIEW

**Comment by rileyorr** [comment]
@scottmries the designs LGTM! However, the UI I saw on `SITEXT` and `PENDNG` were flipped. I wanted to call that out because I wanted to verify that what I'm seeing is correct for each move scenario. 

I've added screenshots of what I noticed with the move code included: 

**PENDNG**
Screenshot 2023-04-27 at 4 19 52 PM

**SITEXT**
Screenshot 2023-04-27 at 4 20 24 PM

**Comment by scottmries** [comment]
@rileyorr Ah, good eye! I updated the comment.

**Review by michaelvaldes-truss — CHANGES_REQUESTED** [review]
Looks great! I added a few small comments, but the feature works overall 👍

File: src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx [hunk 1]
```diff
@@ -47,17 +47,28 @@ const SITHistoryItemHeader = ({ sitItem }) => (
...
+  const sitEndDate = sitStatus
+    ? moment().utc().add(sitStatus?.totalDaysRemaining, 'days')
+    : moment(sitStartDate).utc().add(shipment.sitDaysAllowance, 'days');
+  const sitEndDateElement = <p>{formatDate(sitEndDate, utcDateFormat, 'DD MMM YYYY')}</p>;
 
   // Previous SIT calculations and date ranges
-  const previousDaysUsed = sitStatus.pastSITServiceItems?.map((pastSITItem) => {
-    const sitDaysUsed = moment(pastSITItem.sitDepartureDate).diff(pastSITItem.sitEntryDate, 'days');
+  const previousDaysUsed = sitStatus?.pastSITServiceItems?.map((pastSITItem) => {
+    const sitDaysUsed = moment(pastSITItem.sitDepartureDate).utc().diff(pastSITItem.sitEntryDate, 'days');
```
Comment (afoon) [inline src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx:71]: **question**: Why is `.utc()` needed here? I'm asking cause it was removed in the pass PR cause it was affecting the date math.

File: src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx [hunk 2]
```diff
@@ -47,17 +47,28 @@ const SITHistoryItemHeader = ({ sitItem }) => (
...
+      }
+      return acc;
+    }).sitEntryDate;
+  }
+  const sitStartDateElement = <p>{formatDate(sitStartDate, utcDateFormat, 'DD MMM YYYY')}</p>;
+
+  const sitEndDate = sitStatus
+    ? moment().utc().add(sitStatus?.totalDaysRemaining, 'days')
+    : moment(sitStartDate).utc().add(shipment.sitDaysAllowance, 'days');
+  const sitEndDateElement = <p>{formatDate(sitEndDate, utcDateFormat, 'DD MMM YYYY')}</p>;
```
[the remark this answers is not in the document]
Comment (afoon) [reply src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx:67]: The inclusion the `<p>` on the new date elements is affecting the styles.

File: src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx [hunk 3]
```diff
@@ -66,6 +77,17 @@ const SitStatusTables = ({ shipment, sitExtensions, sitStatus, openModalButton }
 
     return <p key={pastSITItem.id}>{text}</p>;
   });
+
+  // Currently active SIT
+  // const currentLocation = sitStatus.location === LOCATION_TYPES.ORIGIN ? 'origin SIT' : 'destination SIT';
+  let currentLocation = 'origin SIT';
+  if (sitStatus) {
+    currentLocation = sitStatus.location === LOCATION_TYPES.ORIGIN ? 'origin SIT' : 'destination SIT';
+  }
```
Comment (afoon) [inline src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx:86]: **quibble**: Could this work as one ternary?
```suggestion
const currentLocation = sitStatus?.location === LOCATION_TYPES.DESTINATION ? 'destination SIT' : 'origin SIT';
```

**Review by afoon — APPROVED** [review]
Looks good. Just some questions and non-blocking questions

**Review by rileyorr — APPROVED** [review]
@scottmries thank you for confirming the scenarios and updating the comment! Everything LGTM

File: src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (scottmries) [reply src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx]: I removed it.

File: src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx (hunk 1 above)
Comment (scottmries) [reply src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx]: I removed it.

File: src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx (hunk 3 above)
Comment (scottmries) [reply src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx]: Good point. I refactored.

**Pull request merged**